File: PlacementCellBackend/DTOs/InterviewInsights.cs
namespace PlacementCellBackend.DTOs
{
    public class InterviewInsights
    {
        public string CompanyId { get; set; } = string.Empty;
        public string CompanyName { get; set; } = string.Empty;
        public int TotalFeedbacks { get; set; }
        public bool HasCodingRound { get; set; }
        public bool HasTechnicalRound { get; set; }
        public bool HasHRRound { get; set; }
        public List<string> CommonJobProfiles { get; set; } = new();
        public List<string> CommonLocations { get; set; } = new();
        public RoundStatistics RoundStats { get; set; } = new();
    }

    public class RoundStatistics
    {
        public int CodingRoundCount { get; set; }
        public int TechnicalRoundCount { get; set; }
        public int HRRoundCount { get; set; }
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/dtos/InterviewInsights.java
package com.example.placementicsbackend.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class InterviewInsights {

    private String companyId = "";
    private String companyName = "";
    private int totalFeedbacks;
    private boolean hasCodingRound;
    private boolean hasTechnicalRound;
    private boolean hasHRRound;
    private List<String> commonJobProfiles = new ArrayList<>();
    private List<String> commonLocations = new ArrayList<>();
    private RoundStatistics roundStats = new RoundStatistics();
}
